### PR TITLE
feat(App): Loads mode extensions on demand

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

The declaration for the plugin imports file written by writePluginImports didn't allow for the config file to list the extensions/modes by name rather than by hard coded value, causing the initial bundle size to include all plugins. 

### Changes & Results

Split the generated code into bundles by extensions/modes so that they are loaded on demand by string name rather than being incorporated at build time.

### Testing

Load the APP_CONFIG=config/multiple.js to show loading of dev test mode and other extensions.  System should still work as before.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Review by jbocce — COMMENTED** [review]
I ran the `multiple.js` configuration locally and was able to launch the various modes. I assume the non-default extensions loaded fine, but is there a way for me to verify this?

File: modes/basic-dev-mode/src/index.js [hunk 1]
```diff
@@ -183,7 +186,7 @@ function modeFactory({ modeConfiguration }) {
       },
     ],
     extensions: extensionDependencies,
-    hangingProtocol: [ohif.hangingProtocol],
+    hangingProtocol: 'default',
```
Comment (jbocce) [inline modes/basic-dev-mode/src/index.js:189]: Perhaps remove line 13 then?

File: platform/viewer/src/appInit.js [hunk 2]
```diff
@@ -87,25 +93,40 @@ async function appInit(appConfigOrFunc, defaultExtensions, defaultModes) {
...
+    ...(appConfig.modes || []),
+    ...defaultModes,
+  ]);
 
-    const mode = modeFactory(modeConfig);
+  // This is a terrible setup here - todo, change this to use an alternate
+  // name so as to not confuse the issue
+  appConfig.modes = [];
+  // This is a preferred name
+  appConfig.loadedModes = [];
```
[the remark this answers is not in the document]
Comment (wayfarer3130) [reply platform/viewer/src/appInit.js:105]: Removed the update to the modes - now it just applies to loadedModes

**Review by jbocce — APPROVED** [review]

File: platform/viewer/pluginConfig.json [hunk 3]
```diff
@@ -1,48 +1,55 @@
 {
   "extensions": [
+    {
+      "packageName": "@ohif/extension-default"
+    },
     {
       "packageName": "@ohif/extension-cornerstone",
-      "version": "3.0.0"
+      "default": false
```
Comment (sedghi) [inline platform/viewer/pluginConfig.json:8]: why we are getting rid of the version here?

Comment (wayfarer3130) [reply platform/viewer/pluginConfig.json:8]: It is meaningless - only external packages are meaningful to have the version number.

File: platform/public/demo.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<html>
```
Comment (jbocce) [inline platform/public/demo.html:1]: What is this all about?

Comment (wayfarer3130) [reply platform/public/demo.html:1]: It is just a demonstration that one can server up additional files coming from a specified directory.  That allows us to split off static files if we want, and have the directory get included automatically.

File: platform/viewer/pluginConfig.json (hunk 3 above)
Comment (sedghi) [reply platform/viewer/pluginConfig.json:8]: maybe right now is meaningless but in future when we package properly and version them it is needed. So I don't think we remove and add back again, just let it be there

File: platform/public/demo.html (hunk 4 above)
Comment (wayfarer3130) [reply platform/public/demo.html:1]: A place to put demo loads/alternate configs etc.

**Pull request merged**